R/zchunk_LA122.Refining_china.R

```diff
@@ -0,0 +1,238 @@
+    biofuel_MT_province_F %>%
+      gather(year, value, -province, -fuel) %>%
+      filter(fuel == "ethanol") %>%
+      mutate(year = as.numeric(year)) %>%
+      complete(nesting(fuel, province), year = HISTORICAL_YEARS) %>%
+      group_by(province, fuel) %>%
+      mutate( value = as.numeric(value), value = approx_fun(year, value, rule = 1)) %>%
+      ungroup() %>%
+      # ensure all provinces are in the data.frame even thugh they will just be NA
+      full_join(province_names_mappings["province"], by = "province") %>%
+      mutate(fuel = 'corn', sector = "corn ethanol") %>%
+      complete(sector, fuel, province, year = HISTORICAL_YEARS) %>%
+      group_by(year) %>%
+      mutate(value = value / sum(value, na.rm=T)) %>%
+      ungroup() %>%
+      replace_na(list(value = 0)) ->
+      L122.pct_province_btle
+
+    # Corn ethanol output by province
+    L122.pct_province_btle %>%
+      left_join_error_no_match(filter(L122.in_EJ_R_refining_F_Yh, GCAM_region_ID == 11 & sector == "corn ethanol" & fuel == "corn"),
+                by = c("sector", "year")) %>%
+      # province output value = province proportion * national output value
+      mutate(value = value.x * value.y) %>%
+      select(province, sector, fuel = fuel.x, year, value) ->
+      L122.out_EJ_province_btle
+
+    # Corn ethanol inputs by province and fuel: Repeat percentage-wise table by number of fuel inputs
+    # Corn ethanol input fuels
+    L122.in_EJ_R_refining_F_Yh %>%
+      filter(sector == "corn ethanol") %>%
+      select(fuel) %>%
+      distinct() ->
+      corneth_input_fuels
+
+    # Repeat province proportions for all fuels used in corn ethanol sector
+    L122.pct_province_btle %>%
+      select(-fuel) %>%
+      repeat_add_columns(corneth_input_fuels) ->
+    L122.pct_province_btle_repF
+
+    # Corn ethanol inputs by province
+    L122.pct_province_btle_repF %>%
+      left_join_error_no_match(filter(L122.in_EJ_R_refining_F_Yh, GCAM_region_ID == 11), by = c("sector", "fuel", "year")) %>%
+      # province input value = province proportion * national input value
+      mutate(value = value.x * value.y) %>%
+      select(province, sector, fuel, year, value) ->
+      L122.in_EJ_province_btle_F
+
+
+    # Biodiesel output by province
+    # NOTE: SEDS does not cover biodiesel; using a separate EIA database for disaggregating this to provinces
+
+    # Build table of percentages by historical year
+    biofuel_MT_province_F %>%
+      gather(year, value, -province, -fuel) %>%
+      filter(fuel == "biodiesel") %>%
+      mutate(year = as.numeric(year)) %>%
+      complete(nesting(fuel, province), year = HISTORICAL_YEARS) %>%
+      group_by(province, fuel) %>%
+      mutate( value = as.numeric(value), value = approx_fun(year, value, rule = 1)) %>%
+      ungroup() %>%
+      # ensure all provinces are in the data.frame even thugh they will just be NA
+      full_join(province_names_mappings["province"], by = "province") %>%
```

Comment:
Again, this can be done in the `complete` step below: `province = province_names_mappings$province`